scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,204 @@ def visit_compare(self, node):
+                    (python_utils_file, python_utils_test_file,),
+                    ()),
+                (
+                    re.compile(r'\Wnext\('),
+                    b'python_utils.NEXT()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'\Wrange\('),
+                    b'python_utils.RANGE()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'\Wround\('),
+                    b'python_utils.ROUND()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'\Wstr\('),
+                    b'python_utils.convert_to_bytes() ' +
+                    b'or python_utils.UNICODE()',
+                    (python_utils_file,),
+                    ()),
+                (
+                    re.compile(r'\Wzip\('),
+                    b'python_utils.ZIP()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'basestring\('),
+                    b'python_utils.BASESTRING()',
+                    (python_utils_file,),
+                    ()),
+                (
+                    re.compile(r'iteritems\('),
+                    b'items()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'itervalues\('),
+                    b'values()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'iterkeys\('),
+                    b'keys()',
+                    (),
+                    ()),
+            ),
+            'type': 'csv',
+            'metavar': '<comma separated list>',
+            'help': 'List of tuples of disallowed functions '
+                    'and optionally their allowed alternatives.'
+        }),)
+
+    # pylint: enable=disallowed-function-calls,single-line-pragma
+    def _check_exclusion(self, path, excluded_files, excluded_dirs):
+        """Checks if current path should be excluded given a list of
+        excluded files and a list of excluded directories.
+        """
+
+        return any(
+            path.endswith(excluded_file) for excluded_file in excluded_files
+        ) or any(
+            path.startswith(excluded_dir) for excluded_dir in excluded_dirs
+        )
+
+    def visit_call(self, node):
+        """Visit a function call to ensure that the call is
+        not using any disallowed functions.
+
+        Args:
+            node: astroid.nodes.Call. Node to access call content.
+        """
+
+        called = node.as_string()
+        curr_path = node.root().path
+
+        for (
+                func_rgx,
+                replace_msg,
+                excluded_files,
+                excluded_dirs) in self.config.disallowed_functions:
+            if self._check_exclusion(curr_path, excluded_files, excluded_dirs):
+                continue
+            if func_rgx.search(called):
```

Comment:
Instead of regex can we check with the function name directly?

Reply:
Done!